util/dashboard/gen_dashboard_entry.py

```diff
@@ -88,7 +103,68 @@ def gen_dashboard_html(hjson_path, outfile):
                     "        <td>&nbsp;</td>\n")
     genout(outfile, "      </tr>\n")
     # yapf: enable
-    return
+
+
+def print_multiversion_format(obj, outfile):
+    # sort based on the version
+    # TODO: If minor version goes up gte than 10?
+    revisions = sorted(obj["revisions"], key=lambda x: x["version"])
+    outstr = ""
+    for i, rev in enumerate(revisions):
+        outstr += "      <tr>\n"
+        if len(revisions) == 0:
+            outstr += "        <td class='fixleft'>"
+            outstr += html.escape(obj['name']) + "</td>\n"
+        elif i == 0:
+            # First entry
+            outstr += "        <td class='fixleft' rowspan='{}'>".format(
+                len(revisions))
+            outstr += html.escape(obj['name']) + "</td>\n"
+
+        # Version
+        outstr += "        <td class=\"hw-stage\">"
+        outstr += html.escape(rev['version']) + "</td>\n"
+
+        # Life Stage
+        life_stage = rev['life_stage']
+        life_stage_mapping = convert_stage(rev['life_stage'])
+
+        outstr += "        <td class=\"hw-stage\"><span class='hw-stage' title='"
+        outstr += html.escape(life_stage_mapping) + "'>"
+        outstr += html.escape(life_stage) + "</span></td>\n"
+
+        if life_stage != 'L0' and 'design_stage' in rev:
+            design_stage_mapping = convert_stage(rev['design_stage'])
+            outstr += "        <td class=\"hw-stage\"><span class='hw-stage' title='"
+            outstr += html.escape(design_stage_mapping) + "'>"
+            outstr += html.escape(rev['design_stage']) + "</span></td>\n"
+        else:
+            outstr += "        <td>&nbsp;</td>\n"
+
+        if life_stage != 'L0' and 'verification_stage' in rev:
+            verification_stage_mapping = convert_stage(
+                rev['verification_stage'])
+            outstr += "        <td class=\"hw-stage\"><span class='hw-stage' title='"
+            outstr += html.escape(verification_stage_mapping) + "'>"
+            outstr += html.escape(rev['verification_stage']) + "</span></td>\n"
+        else:
+            outstr += "        <td>&nbsp;</td>\n"
+
+        if 'commit_id' in rev:
+            outstr += "        <td>"
+            outstr += "<a href='https://github.com/lowrisc/opentitan/tree/{}'>{}</a>".format(
+                rev['commit_id'], rev['commit_id'][0:7])
+            outstr += "</td>\n"
+        else:
+            outstr += "        <td>&nbsp;</td>\n"
+
+        if 'notes' in rev:
+            outstr += "        <td>" + mk.markdown(rev['notes']) + "</td>\n"
+        else:
+            outstr += "        <td>&nbsp;</td>\n"
+        outstr += "      </tr>\n"
+
+    genout(outfile, outstr)
```

Comment:
This is a lot of HTML generation.  Can you generate markdown instead, and use `{% this %}` style of shortcode to have hugo do the HTML generation?

Reply:
Unfortunately, not possible. `notes` could have multi-lines.

Reply:
Understood.